src/ServiceControl.Audit/Program.cs

```diff
@@ -7,31 +7,40 @@
 using ServiceControl.Configuration;
 using ServiceControl.Infrastructure;
 
-AppDomain.CurrentDomain.UnhandledException += (s, e) => LogManager.GetLogger(typeof(Program)).Error("Unhandled exception was caught.", e.ExceptionObject as Exception);
+try
+{
+    AppDomain.CurrentDomain.UnhandledException += (s, e) => LogManager.GetLogger(typeof(Program)).Error("Unhandled exception was caught.", e.ExceptionObject as Exception);
 
-// Hack: See https://github.com/Particular/ServiceControl/issues/4392
-var exitCode = await IntegratedSetup.Run();
+    // Hack: See https://github.com/Particular/ServiceControl/issues/4392
+    var exitCode = await IntegratedSetup.Run();
 
-if (exitCode != 0)
-{
-    return exitCode;
-}
+    if (exitCode != 0)
+    {
+        return exitCode;
+    }
 
-ExeConfiguration.PopulateAppSettings(Assembly.GetExecutingAssembly());
+    ExeConfiguration.PopulateAppSettings(Assembly.GetExecutingAssembly());
 
-var arguments = new HostArguments(args);
+    var arguments = new HostArguments(args);
 
-if (arguments.Help)
-{
-    arguments.PrintUsage();
-    return 0;
-}
+    if (arguments.Help)
+    {
+        arguments.PrintUsage();
+        return 0;
+    }
 
-var loggingSettings = new LoggingSettings(Settings.SettingsRootNamespace);
-LoggingConfigurator.ConfigureLogging(loggingSettings);
+    var loggingSettings = new LoggingSettings(Settings.SettingsRootNamespace);
+    LoggingConfigurator.ConfigureLogging(loggingSettings);
 
-var settings = new Settings(loggingSettings: loggingSettings);
+    var settings = new Settings(loggingSettings: loggingSettings);
 
-await new CommandRunner(arguments.Command).Execute(arguments, settings);
+    await new CommandRunner(arguments.Command).Execute(arguments, settings);
 
-return 0;
+    return 0;
+}
+finally
+{
+    // Leave a trail in the logs to determine if the process was killed
```

Comment:
Should this be a TODO?

```suggestion
    // TODO Leave a trail in the logs to determine if the process was killed
```